### PR TITLE
feat(blueprint): allow cloning git repo with ref

### PR DESCRIPTION
**Pull request by buschtoens**
This PR adds the ability to checkout a specific ref (branch, tag, commit sha) when using a git repo as the blueprint source, similar to how you can specify named refs as npm dependencies.

```bash
ember new -b https://github.com/ember-cli/app-blueprint-test.git#named-ref
```

It also adds [`--single-branch`](https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---no-single-branch) and [`--depth 1`](https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---depthltdepthgt) to the `git clone` command to speed up the checkout in general.

### CODE REVIEW

File: tests/acceptance/new-test.js [hunk 1]
```diff
@@ -234,6 +234,24 @@ describe('Acceptance: ember new', function() {
...
+    co.wrap(function*() {
+      this.timeout(20000); // relies on GH network stuff
+
+      yield ember([
+        'new',
+        'foo',
+        '--skip-npm',
+        '--skip-bower',
+        '--skip-git',
+        '--blueprint=https://github.com/buschtoens/app-blueprint-test.git#named-ref',
```
Comment (buschtoens) [inline tests/acceptance/new-test.js]:
```suggestion
        '--blueprint=https://github.com/ember-cli/app-blueprint-test.git#named-ref',
```

Pending https://github.com/ember-cli/app-blueprint-test/pull/5 to be merged.

Comment (rwjblue) [reply tests/acceptance/new-test.js]: Merged now...

**Comment by buschtoens** [comment]
Ready for review, in case it slipped your inbox. 😅

**Comment by buschtoens** [comment]
Hmm, good point indeed! I'll revisit the code. I think it shouldn't be too complicated to completely offload this to `yarn` / `npm`.

**Pull request closed without merging**